### PR TITLE
build: updating the rhoas guidelines to the newest version

### PR DESCRIPTION
**Pull request by dimakis**
[Insert JIRA Issue number and update link](https://issues.redhat.com/browse/MGDOBR-)

## PR owner checklist

Updating the OpenAPI Linter to the newest version in the CI. 

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: .github/workflows/openapi_validate.yaml
Comment (jomarko) [inline .github/workflows/openapi_validate.yaml:20]: Could you please explain, why we are not using the latest, i.e. 0.3.1? https://www.npmjs.com/package/@rhoas/spectral-ruleset/v/0.3.1

**Review by jomarko — COMMENTED** [review]
Thank you for the PR @dimakis . One question from me before approval.

File: .github/workflows/openapi_validate.yaml
Comment (dimakis) [reply .github/workflows/openapi_validate.yaml]: great catch, the latest was really just a doc change. cosmetic really. I've removed the explicit version. should pull `latest` now. 👍

**Review by jomarko — APPROVED** [review]

**Comment by jomarko** [comment]
@manstis @danielezonca I think we need one more approval, you were first in my mind. Do you have a minute for review this PR please?

**Review by danielezonca — APPROVED** [review]

**Pull request merged**
